tests/validate/aie_pl_test/src/pl_controller_aie2.hpp

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2022 Xilinx, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#ifndef _XF_PLCTRL_PL_CONTROLLER_AIE2_HPP_
+#define _XF_PLCTRL_PL_CONTROLLER_AIE2_HPP_
+
+//#include "xclbin.h"
+#include <algorithm>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <stdexcept>
+
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/property_tree/ptree.hpp>
+using boost::property_tree::ptree;
+#include "enums.hpp"
+
+namespace xf {
+namespace plctrl {
+
+class dynBuffer {
+   public:
+    uint32_t* data;
+    uint32_t size;
+    uint32_t usedSize;
+    dynBuffer() {
+        data = nullptr;
+        size = 0;
+        usedSize = 0;
+    }
+    uint32_t get(int i) {
+        assert(i < usedSize);
+        return data[i];
+    }
+    void add(uint32_t m_data) {
```

Comment:
Is there a reason why all of these method bodies are in the header file and not body file?

Reply:
I can move these into the body file.  I just took the example code and dropped it in.